### PR TITLE
FEATURE: Group category permissions tab

### PR DESCRIPTION
**Pull request by markvanlan**
Screenshot from 2020-08-06 10-45-27

___
Screenshot from 2020-08-06 10-45-40

### CODE REVIEW

File: app/assets/javascripts/discourse/app/controllers/group-permissions.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import Controller from "@ember/controller";
+
+export default Controller.extend({});
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/controllers/group-permissions.js:3]:
```suggestion
export default Controller.extend();
```

File: app/assets/javascripts/discourse/app/models/permission-type.js [hunk 2]
```diff
@@ -2,23 +2,27 @@ import I18n from "I18n";
 import discourseComputed from "discourse-common/utils/decorators";
 import EmberObject from "@ember/object";
 
+export function buildPermissionDescription(id) {
+  let key = "";
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/models/permission-type.js:6]: This is quite a few more lines than creating a `const PERMISSION_DESCRIPTIONS = { 1: 'full', 2: 'create_post, 3: ... }` then doing `return I18n.t("permission_types." + PERMISSION_DESCRIPTIONS[key]);`

File: app/assets/javascripts/discourse/app/templates/group/permissions.hbs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<section class="user-content">
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/templates/group/permissions.hbs:1]: `user-content` is a weird class name for something related to a group

File: app/controllers/groups_controller.rb [hunk 4]
```diff
@@ -546,6 +546,17 @@ def search
     render_serialized(groups, BasicGroupSerializer)
   end
 
+  def permissions
+    respond_to do |format|
+      format.html { render 'groups/show' }
```
Comment (eviltrout) [inline app/controllers/groups_controller.rb:551]: Curious why this is necessary? By default we bootstrap the app if the request is not an XHR and don't need to render anything specific.

File: config/locales/client.en.yml [hunk 5]
```diff
@@ -682,6 +682,10 @@ en:
           details: "Details"
           from: "From"
           to: "To"
+      permissions:
+        title: "Permissions"
+        none: "No categories specifically permit members of this group to access them."
```
Comment (eviltrout) [inline config/locales/client.en.yml:687]:
```suggestion
        none: "There are no categories associated with this group."
```

File: config/locales/client.en.yml [hunk 6]
```diff
@@ -682,6 +682,10 @@ en:
...
+        description: "Members of this group have access to these categories"
```
Comment (eviltrout) [inline config/locales/client.en.yml:688]:
```suggestion
        description: "Members of this group can access these categories:"
```

File: spec/requests/groups_controller_spec.rb [hunk 7]
```diff
@@ -1733,4 +1733,56 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
...
+    describe "with varying category permissions" do
+      fab!(:category) { Fabricate(:category) }
+
+      before do
+        category.set_permissions("#{group.name}": :full)
+        category.save!
+      end
+
+      it "does not return categories the user cannot see" do
+
```
Comment (eviltrout) [inline spec/requests/groups_controller_spec.rb:1759]: super trivial but I'd remove this blank line.

File: app/assets/javascripts/discourse/app/templates/group/permissions.hbs (hunk 3 above)
Comment (markvanlan) [reply app/assets/javascripts/discourse/app/templates/group/permissions.hbs:1]: This is just the class all the group views use for the section

https://github.com/discourse/discourse/blob/1ca81fbb95ec32fce448c477cbb824e884574c93/app/assets/javascripts/discourse/app/templates/group/messages.hbs#L15

https://github.com/discourse/discourse/blob/1ca81fbb95ec32fce448c477cbb824e884574c93/app/assets/javascripts/discourse/app/templates/group/manage.hbs#L12

https://github.com/discourse/discourse/blob/1ca81fbb95ec32fce448c477cbb824e884574c93/app/assets/javascripts/discourse/app/templates/group/activity.hbs#L13

File: app/controllers/groups_controller.rb (hunk 4 above)
Comment (markvanlan) [reply app/controllers/groups_controller.rb:551]: It must not be :)

File: app/assets/javascripts/discourse/app/models/permission-type.js (hunk 2 above)
Comment (markvanlan) [reply app/assets/javascripts/discourse/app/models/permission-type.js:6]: sounds good. I just moved the function to export it, but it is a good opportunity to refactor :+1:

**Pull request merged**